GitUI/Theming/Win32ThemingHooks.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.Drawing;
+using System.Runtime.InteropServices;
+using EasyHook;
+using GitExtUtils.GitUI.Theming;
+
+namespace GitUI.Theming
+{
+    internal static class Win32ThemingHooks
+    {
+        private static Theme _theme;
+        private static ColorDelegate _colorBypass;
+        private static BrushDelegate _brushBypass;
+        private static MessageBoxADelegate _messageBoxABypass;
+        private static MessageBoxWDelegate _messageBoxWBypass;
+        private static LocalHook _colorHook;
+        private static LocalHook _brushHook;
+        private static LocalHook _messageBoxAHook;
+        private static LocalHook _messageBoxWHook;
+        private static bool _showingMessageBox;
+
+        [UnmanagedFunctionPointer(CallingConvention.StdCall, SetLastError = true)]
+        private delegate int ColorDelegate(int nIndex);
+
+        [UnmanagedFunctionPointer(CallingConvention.StdCall, SetLastError = true)]
+        private delegate IntPtr BrushDelegate(int nIndex);
+
+        [UnmanagedFunctionPointer(CallingConvention.StdCall, SetLastError = true, CharSet = CharSet.Ansi)]
+        private delegate int MessageBoxADelegate(IntPtr hwnd, string test, string caption, int type);
+
+        [UnmanagedFunctionPointer(CallingConvention.StdCall, SetLastError = true, CharSet = CharSet.Unicode)]
+        private delegate int MessageBoxWDelegate(IntPtr hwnd, string test, string caption, int type);
+
+        public static void InstallColorHooks(Theme theme)
+        {
+            (_brushHook, _brushBypass) = InstallHook<BrushDelegate>(
+                "user32.dll",
+                "GetSysColorBrush",
+                BrushHook);
+
+            (_colorHook, _colorBypass) = InstallHook<ColorDelegate>(
+                "user32.dll",
+                "GetSysColor",
+                ColorHook);
+
+            _theme = theme;
+        }
+
+        public static void InstallMessageBoxHooks()
+        {
+            (_messageBoxAHook, _messageBoxABypass) = InstallHook<MessageBoxADelegate>(
+                "user32.dll",
+                "MessageBoxA",
+                MessageBoxAHook);
+
+            (_messageBoxWHook, _messageBoxWBypass) = InstallHook<MessageBoxWDelegate>(
+                "user32.dll",
+                "MessageBoxW",
+                MessageBoxWHook);
+        }
+
+        public static void Uninstall()
+        {
+            _colorHook.Dispose();
```

Comment:
On my development machine, EasyHookDLL is not compiled, so all the hook objects here are null when GE exits. Do you think we need null checking here? Some thing like:
```csharp
            _colorHook?.Dispose();
            _brushHook?.Dispose();
            _messageBoxAHook?.Dispose();
            _messageBoxWHook?.Dispose();
```

Reply:
Is it really feasible to have EasyHook as conditional?
I would prefer to have one setup, which means that C++ will be mandatory.

Reply:
that's right

Reply:
OK, I understand. Do we need to update [build instructions on Wiki](https://github.com/gitextensions/gitextensions/wiki/How-To%3A-build-instructions) to add EasyHook as one of the reasons why we need VS C++, apart from building the installer?

Reply:
Yes when merging this PR.
This may require changes to the installer though, so maybe not as soon as the state of this PR indicates.

Reply:
I don't feel it is necessary to force people to build and use EasyHook locally, if we can avoid it and developers don't desire to do so.